### PR TITLE
Clean useless `.*` in regex

### PR DESCRIPTION
**Pull request by frack113**
### Summary of the Pull Request

- Remove useless `.*` in the regex
- Revert back certain CLI rules to use regex for better accuracy

### Changelog

update: Invoke-Obfuscation CLIP+ Launcher - PowerShell Module - Remove unnecessary starting wildcard
update: Invoke-Obfuscation STDIN+ Launcher - PowerShell Module - Remove unnecessary starting wildcard
update: Invoke-Obfuscation VAR+ Launcher - PowerShell Module - Remove unnecessary starting wildcard
update: Invoke-Obfuscation Via Stdin - PowerShell Module - Remove unnecessary starting wildcard
update: Invoke-Obfuscation Via Use Clip - PowerShell Module - Remove unnecessary starting wildcard
update: Invoke-Obfuscation VAR++ LAUNCHER OBFUSCATION - PowerShell Module - Remove unnecessary starting wildcard
update: Invoke-Obfuscation CLIP+ Launcher - PowerShell - Remove unnecessary starting wildcard
update: Invoke-Obfuscation STDIN+ Launcher - Powershell - Remove unnecessary starting wildcard
update: Invoke-Obfuscation VAR+ Launcher - PowerShell - Remove unnecessary starting wildcard
update: Invoke-Obfuscation Via Stdin - Powershell - Remove unnecessary starting wildcard
update: Invoke-Obfuscation Via Use Clip - Powershell - Remove unnecessary starting wildcard
update: Invoke-Obfuscation VAR++ LAUNCHER OBFUSCATION - PowerShell - Remove unnecessary starting wildcard
update: Invoke-Obfuscation STDIN+ Launcher - Update rule to use regex for better accuracy in CLI
update: Invoke-Obfuscation VAR+ Launcher - Update rule to use regex for better accuracy in CLI
update: Invoke-Obfuscation Via Stdin - Update rule to use regex for better accuracy in CLI
update: Invoke-Obfuscation Via Use Clip - Update rule to use regex for better accuracy in CLI

### Example Log Event
N/A

### Fixed Issues
N/A

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/sigmahq_conventions.md)

### CODE REVIEW

**Review by phantinuss — DISMISSED** [review]

**Review by nasbench — APPROVED** [review]

**Comment by nasbench** [comment]
- Reverted back the deleted comments @frack113 as they were there in case we wanted to revert back to regex.
- I reverted some CLI rules to use regex as its much more accurate and i don't see why we don't keep them as regex anymore. Changing them to contains is basically "regex" for a lot of EDR/SIEMs

@phantinuss can you double check before i merge.

**Comment by phantinuss** [comment]
Plus of the regex is that the order or arguments is relevant, wheres with contains it's not. Can be upside or downside depending on the specific case.

File: rules/windows/process_creation/proc_creation_win_hktl_invoke_obfuscation_stdin.yml [hunk 1]
```diff
@@ -16,22 +16,11 @@ logsource:
...
-        CommandLine|contains:
-            - '/c'
-            - '/r'
-    selection_other:
-        - CommandLine|contains: 'noexit'
-        - CommandLine|contains|all:
-              - 'input'
-              - '$'
-    condition: all of selection_*
+        CommandLine|re: 'cmd.{0,5}(?:\/c|\/r).+powershell.+(?:\$\{?input\}?|noexit).+\"'
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_hktl_invoke_obfuscation_stdin.yml:22]:
```suggestion
        CommandLine|re: '(?i)cmd.{0,5}(?:\/c|\/r).+powershell.+(?:\$\{?input\}?|noexit).+\"'
```
don't know out of my head but I think `|re` should be case sensitive and even if not we should add it for consistency when looking at the other regexes

File: rules/windows/process_creation/proc_creation_win_hktl_invoke_obfuscation_var.yml [hunk 2]
```diff
@@ -17,16 +17,9 @@ logsource:
...
         # Example 1: C:\winDoWs\SySTeM32\cmd.Exe /C"SET NOtI=Invoke-Expression (New-Object Net.WebClient).DownloadString&& PowERshElL -NOl SET-iteM ( 'VAR' + 'i'+ 'A' + 'blE:Ao6' + 'I0') ( [TYpe](\"{2}{3}{0}{1}\"-F 'iRoN','mENT','e','nv') ) ; ${exECUtIONCOnTEXT}.\"IN`VO`KecOmMaND\".\"inVo`KES`crIPt\"( ( ( GEt-VAriAble ( 'a' + 'o6I0') -vaLU )::(\"{1}{4}{2}{3}{0}\" -f'e','gETenvIR','NtvaRIa','BL','ONme' ).Invoke(( \"{0}{1}\"-f'n','oti' ),( \"{0}{1}\" -f'pRoC','esS') )) )"
         # Example 2: cMD.exe /C "seT SlDb=Invoke-Expression (New-Object Net.WebClient).DownloadString&& pOWErShell .(( ^&(\"{1}{0}{2}{3}\" -f 'eT-vaR','G','iab','lE' ) (\"{0}{1}\" -f '*m','DR*' ) ).\"na`ME\"[3,11,2]-JOIN'' ) ( ( ^&(\"{0}{1}\" -f'g','CI' ) (\"{0}{1}\" -f 'ENV',':SlDb' ) ).\"VA`luE\" ) "
-        CommandLine|contains|all:
-            - 'cmd'
-            - '"set'
-            - '-f'
-        CommandLine|contains:
-            - '/c'
-            - '/r'
+        CommandLine|re: 'cmd.{0,5}(?:\/c|\/r)(?:\s|)\"set\s[a-zA-Z]{3,6}.*(?:\{\d\}){1,}\\\"\s+?\-f(?:.*\)){1,}.*\"'
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_hktl_invoke_obfuscation_var.yml:22]:
```suggestion
        CommandLine|re: 'cmd.{0,5}(?:\/c|\/r)\s?\"set\s[a-zA-Z]{3,6}.*(?:\{\d\}){1,}\\\"\s+?\-f(?:.*\)){1,}.*\"'
```

Comment (phantinuss) [reply rules/windows/process_creation/proc_creation_win_hktl_invoke_obfuscation_var.yml:22]: We shouldn't "abuse" groups with alternates for optional matching.

File: rules/windows/process_creation/proc_creation_win_hktl_invoke_obfuscation_via_use_clip.yml [hunk 3]
```diff
@@ -17,22 +17,9 @@ logsource:
...
-        CommandLine|contains:
-            - 'clipboard'
-            - 'invoke'
-            - 'i`'
-            - 'n`'
-            - 'v`'
-            - 'o`'
-            - 'k`'
-            - 'e`'
+        CommandLine|re: '(?i)echo.*clip.*&&.*(Clipboard|i`?n`?v`?o`?k`?e`?).*'
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_hktl_invoke_obfuscation_via_use_clip.yml:22]:
```suggestion
        CommandLine|re: '(?i)echo.*clip.*&&.*(Clipboard|i`?n`?v`?o`?k`?e`?)'
```
`.*`s in the end also not needed

**Review by phantinuss — APPROVED** [review]

File: rules/windows/process_creation/proc_creation_win_hktl_invoke_obfuscation_via_use_clip.yml (hunk 3 above)
Comment (phantinuss) [reply rules/windows/process_creation/proc_creation_win_hktl_invoke_obfuscation_via_use_clip.yml]: pushed a fix for other affected files as well

File: rules/windows/process_creation/proc_creation_win_hktl_invoke_obfuscation_stdin.yml (hunk 1 above)
Comment (nasbench) [reply rules/windows/process_creation/proc_creation_win_hktl_invoke_obfuscation_stdin.yml]: Yes it is case sensitive and we had a discussion around this. Hence the creation of the new modifier `i` and other specifically to be combined with regex.

**Pull request merged**